azext_iot/tests/deviceupdate/conftest.py

```diff
@@ -308,6 +309,25 @@ def _iothub_provisioner(request) -> Optional[dict]:
         if desired_instance_count:
             hub_id_map = {}
             hub_names = []
+
+            # Assign Data Contributor role to resource group if it does not exist
+            account = cli.invoke("account show").as_json()
+            scope_id = "/subscriptions/{}/resourceGroups/{}".format(
```

Comment:
1. Should we be lazy and do this for all services? For other tests, we assign permissions one by one but if we can just add all the permissions needed to the rg and not worry about them after, it would be faster. 

But this idea is also a bad security practice and may end up with extra assignments that are not cleaned up in the end

2. Should we try to clean up resource permission assignments - as you got permission for x for a test, do we want to remove it post test? This sounds very annoying though.

@c-ryan-k  thoughts?